### PR TITLE
Document Generator update exception handling

### PR DESCRIPTION
**Pull request by Mathsh**
Updated the exception handling within document-generator to enable exceptions to be passed back to the api from the different implementations

- Added new exception wrapper to document-generator-interfaces
- Updated ResponseStatus to include additional responses for accounts
- Updated ApiMapper to handle new responseStatus
- Updated JUnit tests to for changes
- Created new function to Create and Log error messages
- Updated document-generator-accounts to wrap error to DocumentGeneratorInterfaceException to pass back

Resolves SFA - 723

### CODE REVIEW

File: document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java [hunk 1]
```diff
@@ -29,18 +33,22 @@
     private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
 
     @Override
-    public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) {
+    public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) throws DocumentGeneratorInterfaceException {
```
Comment (jsimons1989) [inline document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java]: The exception doesn't feel right. The interface itself isn't throwing this exception. Maybe DocumentInfoException would be more related to the issue which caused the exception?

File: document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java [hunk 2]
```diff
@@ -65,21 +75,30 @@ void testUnsuccessfulGetDocumentInfoNoAccountsResourceInTransaction() throws Ser
...
 
         assertThrows(HandlerException.class, () -> accountsHandlerMock.getAbridgedAccountsData(transaction, ""));
-
-        assertNull(accountsDocumentInfoService.getDocumentInfo(createDocumentInfoRequest()));
     }
 
     @Test
     @DisplayName("Tests the successful retrieval of document data")
-    void testSuccessfulGetDocumentInfo() throws HandlerException, ServiceException {
+    void testSuccessfulGetDocumentInfo() throws HandlerException, ServiceException, DocumentGeneratorInterfaceException {
```
Comment (jsimons1989) [inline document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java]: This shouldn't need to catch exceptions from 3 different layers.

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java [hunk 3]
```diff
@@ -67,50 +68,47 @@ public DocumentGeneratorServiceImpl(DocumentInfoServiceFactory documentInfoServi
...
-                    .get(documentType.toString())
-                    .getDocumentInfo(documentInfoRequest);
-
+        DocumentInfoResponse documentInfoResponse;
+        try {
+            documentInfoResponse = documentInfoServiceFactory
+                        .get(documentType.toString())
+                        .getDocumentInfo(documentInfoRequest);
+        } catch (DocumentGeneratorInterfaceException dgie) {
+             createAndLogErrorMessage("Error occurred whilst obtaining the data to generator document " +
```
Comment (jsimons1989) [inline document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java]: Typo in the error message

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java [hunk 4]
```diff
@@ -178,4 +176,33 @@ private DocumentResponse setDocumentResponse(RenderDocumentResponse renderRespon
...
+     * @param resourceId The resource Id
+     * @param resourceUri The resource Url
+     * @param requestId The request Id
+     */
+    private <T extends Exception> void createAndLogErrorMessage(String message, T exception,
+                                                                String resourceId, String resourceUri, String requestId) {
+
+        Map < String, Object > debugMap = new HashMap < > ();
+        debugMap.put("resource_uri", resourceUri);
+        debugMap.put("resource_id",resourceId);
```
Comment (jsimons1989) [inline document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java]: Given these keys are used in different classes they could be placed in shared constants or an enum to ensure consistency between all the logs?

File: document-generator-api/src/test/java/uk/gov/companieshouse/document/generator/api/service/DocumentGeneratorServiceTest.java [hunk 5]
```diff
@@ -79,7 +80,7 @@ public void setUp() {
 
     @Test
     @DisplayName("Test a successful generate completed")
-    public void testsSuccessfulGenerateCompleted() throws IOException, DocumentGeneratorServiceException {
+    public void testsSuccessfulGenerateCompleted() throws IOException, DocumentGeneratorServiceException, DocumentGeneratorInterfaceException {
```
Comment (jsimons1989) [inline document-generator-api/src/test/java/uk/gov/companieshouse/document/generator/api/service/DocumentGeneratorServiceTest.java]: Shouldn't this only be returning one exception type?

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java (hunk 4 above)
Comment (Mathsh) [reply document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java]: I have a story to refine the logging as I need to go over the details with yourself.

File: document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java (hunk 2 above)
Comment (Mathsh) [reply document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java]: yeah the handler and service one are needed for;

` when(transactionService.getTransaction(anyString())).thenReturn(createTransaction());
   when(accountsHandlerMock.getAbridgedAccountsData(any(Transaction.class), anyString())).thenReturn(new DocumentInfoResponse());`

File: document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java (hunk 4 above)
Comment (Mathsh) [reply document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/impl/DocumentGeneratorServiceImpl.java]: SFA-722

**Pull request merged**